docs/testing.md

```diff
@@ -1,54 +1,126 @@
----
-next: docs/logging.md
----
-
```

Comment:
Did you intend to delete this? Is it obsolete?

Reply:
This is necessary

Reply:
I must have lost it somewhere in between copy pasting the text. I'll add it back!